lib/macos.js

```diff
@@ -24,49 +117,18 @@ var MacosBuilder = {
    */
   build : function( options, callback ) {
 
-    if ( process.platform !== 'darwin' ) return callback( new Error( 'Invalid platform.' ) )
-
     options.log( '- Starting build for ´' + options.platform + '´ - ' );
 
-    options.log( 'Writing temporary ´appdmg.json´' );
-
-    options.config.macos.background         = path.join( options.basePath, options.config.macos.background );
-    options.config.macos.icon               = path.join( options.basePath, options.config.macos.icon );
-    options.config.macos.contents[ 1 ].path = options.appPath;
-
-    var configFilePath = path.join( os.tmpDir(), 'appdmg.json' );
-
-    fs.writeFileSync(
-      configFilePath,
-      JSON.stringify( options.config.macos ),
-      {
-        encoding : 'utf8'
-      }
-    );
-
-    options.log( 'Wrote temporary ´appdmg.json´' );
-    options.log( 'Kicking off ´appdmg´' );
-
-    var appdmg = require( 'appdmg' );
-    var ee = appdmg( {
-      source : configFilePath,
-      target : path.join( options.out, options.config.macos.title + '.dmg' )
-    } );
-
-    ee.on( 'progress', function ( info ) {
-      if ( info.type === 'step-begin' ) {
-        options.log( 'appdmg: [' + info.current + '] ' + info.title  );
-      }
-    } );
-
-    ee.on('finish', function () {
-      options.log( 'Finished ´appdmg´' );
-      callback();
-    } );
-
-    ee.on( 'error', function ( error ) {
-      callback( error );
-    } );
```

Comment:
This entire red block was simply moved up above, verbatim, into the function `buildDarwin`.